python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -113,6 +114,20 @@ class ComponentTypeMetadata(ComponentTypeInternalMetadata):
     package: str
 
 
+@record
+class ComponentTypeKey:
+    name: str
+    package: str
```

Comment:
I think we need to bikeshed here. 

Elsewhere we use the term "component name" to refer to the full string but here name is only the local name. We need to come up with consitent nomenclature.